src/components/DateSelector.vue

```diff
@@ -0,0 +1,75 @@
+<template>
+  <div>
+    <va-date-input
+    label="due date"
+    v-model="value"
+    :format="formatDate"
+    :parse="parseFn"
+    placeholder="yyyy-mm-dd"
+    clearable
+    manual-input
+  />
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, ref, watch, PropType, onMounted } from "vue";
+import { Task } from "@/types/asana";
+import { format, isValid, parse } from "date-fns";
+import store from "@/store";
+
+export default defineComponent ({
+  props: {
+    task: {
+      type: Object as PropType<Task>,
+      required: true
+    }
+  },
+  setup(props) {
+    const value = ref<Date>();
+
+    const formatDate = (date: Date | undefined) => {
+
+      if (isValid(date)) {
+        return format(date!, "yyyy-MM-dd");
+      }
+
+      if (date === undefined) {
+        return ""
+      }
+
+      return "Invalid Date"
+    };
+
+    const parseFn = (event: Event) => {
+      const dateString = (event.target as HTMLInputElement).value
+      const [year, month, day] = dateString.split('-')
```

Comment:
This split is tied directly to the format convention. I think best to try and find a library to do this for us. Does parse() from date-fns work?